### PR TITLE
chore: Minor improvements to address change requests to list vault on charmhub

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

This PR addresses the change requests made in #472 

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ghislainbourgeois — APPROVED** [review]

File: lib/charms/vault_k8s/v0/vault_autounseal.py [hunk 1]
```diff
@@ -272,9 +283,11 @@ def __init__(self, charm: ops.CharmBase, relation_name: str):
         )
 
     def _on_relation_created(self, event: ops.RelationCreatedEvent) -> None:
+        """Handle the relation created event and emit a custom event."""
```
Comment (arturo-seijas) [inline lib/charms/vault_k8s/v0/vault_autounseal.py:286]: You might want to document the ' event` arg here

File: lib/charms/vault_k8s/v0/vault_autounseal.py [hunk 2]
```diff
@@ -272,9 +283,11 @@ def __init__(self, charm: ops.CharmBase, relation_name: str):
...
         self.on.vault_autounseal_requirer_relation_created.emit(relation=event.relation)
 
     def _on_relation_broken(self, event: ops.RelationBrokenEvent) -> None:
+        """Handle the relation broken event and emit a custom event."""
```
Comment (arturo-seijas) [inline lib/charms/vault_k8s/v0/vault_autounseal.py:290]: You might want to document the ' event` arg here

File: lib/charms/vault_k8s/v0/vault_autounseal.py (hunk 1 above)
Comment (saltiyazan) [reply lib/charms/vault_k8s/v0/vault_autounseal.py:357]: I'm not sure that would add any value.
I'd probably write something like:

```
Args:
   event: (RelationCreatedEvent) Triggered by Juju when event is created

```

Comment (arturo-seijas) [reply lib/charms/vault_k8s/v0/vault_autounseal.py:286]: Well, it's up to you and the conventions you are following. I'm just asking :)

**Review by arturo-seijas — APPROVED** [review]

**Review by ghislainbourgeois — APPROVED** [review]

**Pull request merged**